airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/destination_async/DetectStreamToFlushTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination_async;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.destination_async.buffers.BufferDequeue;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class DetectStreamToFlushTest {
+
+  public static final Instant NOW = Instant.now();
+  public static final Instant FIVE_MIN_AGO = NOW.minusSeconds(60 * 5);
+  private static final long SIZE_10MB = 10 * 1024 * 1024; // 10MB
+  private static final StreamDescriptor DESC1 = new StreamDescriptor().withName("test1");
+  private static final StreamDescriptor DESC2 = new StreamDescriptor().withName("test2");
+  private static final Set<StreamDescriptor> DESCS = Set.of(DESC1, DESC2);
+
+  @Nested
+  class Threshold {
```

Comment:
I added grouping by test categories because I felt it made it easier to read when as I reviewed locally. feel free to remove if you don't like it

Reply:
`@Nested` frustrates me mostly because intellij can't handle them well.

Reply:
in the next PR, i'll split these up into separate classes to be easier to consume